src/components/ec-input-field/ec-input-field.vue

```diff
@@ -1,14 +1,30 @@
 <template>
-  <div class="ec-input-field">
+  <div
+    class="ec-input-field"
+    data-test="ec-input-field"
+  >
     <label
-      v-if="label"
+      v-if="label || note"
       class="ec-input-field__label"
+      data-test="ec-input-field__label"
       :for="id"
-    >{{ label }}</label>
+    >
+      <span
+        v-if="label"
+        class="ec-input-field__label-text"
+        data-test="ec-input-field__label-text"
+      >{{ label }}</span>
+      <span
+        v-if="note"
+        class="ec-input-field__note"
+        data-test="ec-input-field__note"
+      >{{ note }}</span>
+    </label>
     <input
       :id="id"
       v-model="inputModel"
       class="ec-input-field__input"
+      :data-test="$attrs['data-test'] ? $attrs['data-test'] : 'ec-input-field__input'"
```

Comment:
I don't know if this is okay or not, but I couldn't find another solution to the problem.

The problem is, for example in the dropdown we set a data-test on the ec-input-field and we want to have this data-test on the input of the ec-input-field, if we set the data-test to this input without any validation we never gonna get the data test from the parent.

Reply:
First of all, we need to preserve both, so it should be `${$attrs['data-test']} ec-input-field__input`, then we need to create something more reusable, because we are going to have the same problem with every component, e.g.

```
<ec-alert data-test="my-alert" />
```
^^ will override `data-test="ec-alert"` set inside of `ec-alert.vue`. In previous project I used a directive v-merge-data-test="'ec-alert'", which can merge everything the same way. It's not hard to implement, we just need to catch all cases, e.g. multiple nested components:
```
<ebo-suppliers-dropdown data-test="one" />
contains in template:
<ec-dropdown data-test="two" />
contains in template:
<ec-input-field data-test="three" />
```
should output `<input data-test="one two three" >`